packages/block-library/src/comment-template/edit.js

```diff
@@ -124,18 +124,29 @@ export default function CommentTemplateEdit( {
 		( select ) => {
 			const { getEntityRecords } = select( coreStore );
 			const { getBlocks } = select( blockEditorStore );
-
+			const commentQuery = {
+				post: postId,
+				status: 'approve',
+				context: 'embed',
+			};
+			// We cannot get the order of the discussion settings page.
+			// Without the order defined, the default on the API is DESC
+			// with means show the newest comments first.
+			// By default, in discussion settings, the order is ASC.
+			// So the frontend is showing older comments first, while the editor is showing newer ones.
+			if ( order ) {
+				commentQuery.order = order;
+			}
```

Comment:
I'm not confident with this solution, as both editor and frontend differ by default.

What is your opinion about this? 

Could we solve it with the `toggle box` of `inherit from settings` proposal?

That way, the order would be:
 - If inherit enabled: settings
 - If inherit disabled:  DESC (newer first) by default like the REST API, otherwise chosen by the user with the toggle.

cc: @gziolo @ntsekouras @jasmussen @michalczaplinski @DAreRodz

Reply:
We should read the default values from the server and use them when there is no explicit override selected by the user. I think @michalczaplinski at some point had some code that exposes the site settings through the REST API. Am I correct or was it something else?

Reply:
Yes! I wanted to have `comments_per_page` exposed in the REST API. Here are quite complete instructions on how to do it, they should apply to the `order` property as well: 
https://wordpress.slack.com/archives/C02QB2JS7/p1635376326175900

For those without access to the WP slack, here's how it can be done. It should be added to `/lib/rest-api.php`
```php
function register_block_core_comment_template_order_setting() {
  register_setting(
	  'discussion',
	  'order',
	  array(
		  'show_in_rest' => true,
	  )
  );
}

add_action( 'rest_api_init', 'register_block_core_comment_template_order_setting', 10 );
```

Reply:
Wow, this is a really quick way to fix it, it will save me a lot of time 🎉 

Would it be a good idea to create an endpoint to retrieve all the discussion settings? We could use `register_rest_route`, get all Discussion settings and use them for all the block rendering.

Reply:
I don't think you need to create a new endpoint for that. As far as I'm aware the code that @michalczaplinski shared, adds the setting to the endpoint that is already in use in Gutenberg.

Reply:
I have an update in regards to these settings. I talked to @spacedmonkey and he explained that the https://make.wordpress.org/core/wp-json/wp/v2/settings endpoint is only available for admins. There is a long-standing Tract ticket that we would need to land: https://core.trac.wordpress.org/ticket/48885 – the idea is to expose site settings in the read-only mode for all user roles.